Lukem/index page learn (#212)
moondance-labs/tanssi-docs: learn/framework/index.md

---
title: Appchain Development Framework
description: Learn about the appchain Development Framework, Rust Programming Language, Substrate and its features, including XCM and ready-to-use pallets for your Runtime.
template: index-page.html
---

# Appchain Development Framework

> Building a blockchain from scratch is an intricate process, demanding deep knowledge of consensus algorithms, cryptography, data structures, governance, and more. With Tanssi, you can bypass these complexities and focus entirely on your vision. The Appchain Development Framework offers a streamlined approach to creating tailored blockchains, leveraging Substrate's powerful architecture to handle every essential internal component of a blockchain.
> 
> Designed for flexibility, this framework allows you to adapt runtime characteristics, integrate pre-built modules, or create new ones based on your appchain’s specific needs. Whether it’s enhancing core functionality, optimizing for scalability, or enabling seamless user interactions, Tanssi provides the foundation to bring your blockchain ideas to life.

Reviewer: ```suggestion
Building a blockchain from scratch is an intricate process that demands deep knowledge of consensus algorithms, cryptography, data structures, governance, and more. Tanssi simplifies this process by leveraging Substrate, a powerful blockchain development framework that provides all the essential components for creating customized blockchains.

With Substrate’s modular architecture, you can modify runtime characteristics, integrate pre-built modules, or develop new ones based on your appchain’s specific needs. Tanssi provides the foundation you need to bring your blockchain vision to life.
```

Key Features of the Framework:

> - **Substrate Core Architecture:** Divided into the **Core Client**, which manages external communications and internal responsibilities like storage, and the **Runtime**, which executes transactions and manages state transitions with customizable logic

Reviewer: Here, we don't need to put Core Client and Runtime in bold. In most cases, we'll use italics to draw attention to specific words and we leave bold for things that are super important, must-follow type things where we're sort of like HEY! PAY ATTENTION! haha

Also, no need to capitalize either of these words. The only time we really capitalize a word mid-sentence is like in link text (but only when the link text is the name of the page that we're linking to), or if it's a product name and always gets capitalized

> - **End-User Simplicity:** The internal workings of the framework are abstracted for end-users, ensuring seamless interaction through dApps or RPC endpoints, such as wallets.

Reviewer: ```suggestion
- **End-User Simplicity:** The internal workings of the framework are abstracted for end-users, ensuring seamless interaction through dApps or via RPC endpoints, such as those used by wallets
```

I feel like this minor distinction helps to separate dapps and RPC endpoints

---
title: Appchain Development Framework
description: Learn about the appchain Development Framework, Rust Programming Language, Substrate and its features, including XCM and ready-to-use pallets for your Runtime.
icon: octicons-code-24
template: index-page.html
---

> # Appchain Development Framework

Reviewer: ```suggestion
# Network Development Framework
```

Please check the rest of this page and switch appchain > network, including the title and description at the top of the page

Building a blockchain from scratch is an intricate process that demands deep knowledge of consensus algorithms, cryptography, data structures, governance, and more. Tanssi simplifies this process by leveraging Substrate, a powerful blockchain development framework that provides all the essential components for creating customized blockchains.

With Substrate’s modular architecture, you can modify runtime characteristics, integrate pre-built modules, or develop new ones based on your appchain’s specific needs. Tanssi provides the foundation you need to bring your blockchain vision to life.

> Key Features of the Framework:

Reviewer: ```suggestion
Key Features of the framework:
```

> Key Features of the framework:

Reviewer: ```suggestion
Key features of the framework:
```

> - **Substrate core architecture** - divided into the *Core Client*, which manages external communications and internal responsibilities like storage, and the *Runtime*, which executes transactions and manages state transitions with customizable logic

Reviewer: So, this one is kind of weird because there are multiple ways to add bold and italics styling, but we try to stick to the following conventions:

Bold - use `**something in bold**` (instead of `__something in bold__`)
Italics - use `_something in italics_` (instead of `*something in italics*`)

Reviewer: Also, we don't need to capitalize Core Client or Runtime here

> - **Substrate core architecture** - divided into the _core client_, which manages external communications and internal responsibilities like storage, and the _Runtime_, which executes transactions and manages state transitions with customizable logic

Reviewer: ```suggestion
- **Substrate core architecture** - divided into the _core client_, which manages external communications and internal responsibilities like storage, and the _runtime_, which executes transactions and manages state transitions with customizable logic
```

> - **End-User simplicity** - the internal workings of the framework are abstracted for end-users, ensuring seamless interaction through dApps or via RPC endpoints, such as those used by wallets

Reviewer: ```suggestion
- **End-user simplicity** - the internal workings of the framework are abstracted for end-users, ensuring seamless interaction through dApps or via RPC endpoints, such as those used by wallets
```

- **Modular system design** - incorporate pre-built modules or create custom ones to meet specific needs

> - **Developer-centric flexibility** - focus on runtime characteristics and use case specifics, allowing you to create a blockchain tailored to your vision without unnecessary complexity.

Reviewer: ```suggestion
- **Developer-centric flexibility** - focus on runtime characteristics and use case specifics, allowing you to create a blockchain tailored to your vision without unnecessary complexity
```
